packages/java/engine-core/src/main/java/dev/hilla/engine/EngineConfiguration.java

```diff
@@ -3,47 +3,67 @@
+import java.util.Set;
+import java.util.stream.Collectors;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import com.fasterxml.jackson.annotation.PropertyAccessor;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 
 public class EngineConfiguration {
-    public static final String RESOURCE_NAME = "hilla-engine-configuration.json";
-
+    public static final String DEFAULT_CONFIG_FILE_NAME = "hilla-engine-configuration.json";
     public static final String OPEN_API_PATH = "dev/hilla/openapi.json";
-
-    private static final ObjectMapper MAPPER = new ObjectMapper()
+    static final ObjectMapper MAPPER = new ObjectMapper()
             .setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE)
             .setVisibility(PropertyAccessor.FIELD,
                     JsonAutoDetect.Visibility.ANY);
-
     private Path baseDir;
-    private LinkedHashSet<String> classPath;
+    private Path buildDir;
+    @JsonDeserialize(as = LinkedHashSet.class)
+    private Set<Path> classPath;
+    private Path classesDir;
     private GeneratorConfiguration generator;
+    private Path outputDir;
     private ParserConfiguration parser;
-    private String buildDir;
-    private String classesDir;
 
-    private String outputDir = "frontend/generated";
+    private EngineConfiguration() {
+    }
 
     /**
-     * Reads the configuration from the given base directory.
+     * Reads the configuration from the given base directory. Reads only files
+     * with the default name.
      *
-     * @param targetDir
-     *            the directory where the configuration file is expected to be
+     * @param configDir
+     *            a directory that contains the configuration file.
      * @return the configuration, or <code>null</code> if the configuration file
      *         does not exist
      * @throws IOException
      *             if thrown while reading the configuration file
      * @throws ConfigurationException
      *             if the configuration file is invalid
      */
-    public static EngineConfiguration load(File targetDir) throws IOException {
-        File configFile = new File(targetDir, RESOURCE_NAME);
+    public static EngineConfiguration loadDirectory(Path configDir)
+            throws IOException {
+        return load(configDir.resolve(DEFAULT_CONFIG_FILE_NAME).toFile());
+    }
 
+    /**
+     * Reads the configuration from the given file path.
+     *
+     * @param configFile
+     *            a path to a configuration file.
+     * @return the configuration, or <code>null</code> if the configuration file
+     *         does not exist
+     * @throws IOException
+     *             if thrown while reading the configuration file
+     * @throws ConfigurationException
+     *             if the configuration file is invalid
+     */
+    public static EngineConfiguration load(File configFile) throws IOException {
```

Comment:
Since we don't intent to make the file name configurable, should this be a private method?